tRPC: added a step to create the context with every query (#612)
clerk/clerk-docs: docs/references/nextjs/trpc.mdx

export type Context = trpc.inferAsyncReturnType<typeof createContext>;
```

> Then, add the Clerk context to every tRPC query sent to the server:

Reviewer: ```suggestion
Next step, add the Clerk context to every tRPC query sent to the server:
```

Reviewer 2: I think the first way sounds better. 
We would write "Next step" if it was in another step, but in this case, it is not. It's just another instruction in the same step.

Reviewer 3: I would drop the 'Next step' or 'Then' and just do 'Add the Clerk....'

```ts filename="src/pages/api/trpc/[trpc].ts"
import { appRouter } from '@/server/routers'
import * as trpcNext from '@trpc/server/adapters/next'
import { createContext } from 'src/server/context';

export default trpcNext.createNextApiHandler({
  router: appRouter,
  createContext: createContext,
});
```

> ## Accessing the context data in your back-end

Reviewer 2: Following styleguide rule [2.4 Avoid gerunds](https://github.com/clerk/clerk-docs/blob/main/STYLEGUIDE.md#24-avoid-gerunds--ing-words)

```suggestion
## Access the context data in your backend
```

> Once you have context in your back-end, you can access the data in any procedure by using `ctx` in your queries.

Reviewer: ```suggestion
Once you have context in your backend, you can access the data in any procedure by using `ctx` in your queries.
```

})
```

> ## Update your tRPC Middleware

Reviewer: ```suggestion
## tRPC Middleware
```

Reviewer: change related to one below

Reviewer 2: Actually, I'd change it to `Create a Protected Procedure`
see my notes one below

> In most cases, you need to protect your routes from users who aren't signed in. You can create a protected procedure by updating your tRPC middleware like so:

Reviewer: ```suggestion

In many applications, it's essential to restrict access to certain routes based on user authentication status. This ensures that sensitive data and functionality are only accessible to authorized users. tRPC middleware provides a powerful mechanism for implementing this protection within your application.

### Creating a Protected Procedure:

Follow these steps to create a procedure that requires user authentication:
```
@alexisintech thoughts on the rewording?

Reviewer 2: I'd change the heading to `Update your tRPC middleware` and the heading above this one to read `Create a Protected Procedure`. Because updating your tRPC middleware is a step within creating a protected procedure.

and then instead of the "Follow these steps" line, which is a little vague, I'd write:

To create a protected procedure, you first need to update your tRPC middleware. In the example below, the `isAuthed` helper uses the context (`ctx`) to get access to the user's ID. If the user's ID exists in the authentication context, this means that the user is signed in. If the user is not signed in, an `UNAUTHORIZED` error is thrown.

> Once you have created your procedure, you can use it in any router. The example below uses the protected procedure and returns the user's name.

Reviewer 2: ```suggestion
The example below uses the protected procedure and returns a secret message if the user is signed in.
```
```suggestion
Once you have created your procedure, you can use it in any router. The example below uses the protected procedure and returns a secret message if the user is signed in.
```
